File: DynamicKey/AgoraDynamicKey/java/src/main/java/io/agora/sample/FpaTokenBuilderSample.java
package io.agora.sample;

import io.agora.media.FpaTokenBuilder;

public class FpaTokenBuilderSample {
    // Need to set environment variable AGORA_APP_ID
    static String appId = System.getenv("AGORA_APP_ID");
    // Need to set environment variable AGORA_APP_CERTIFICATE
    static String appCertificate = System.getenv("AGORA_APP_CERTIFICATE");

    public static void main(String[] args) {
        System.out.printf("App Id: %s\n", appId);
        System.out.printf("App Certificate: %s\n", appCertificate);
        if (appId == null || appId.isEmpty() || appCertificate == null || appCertificate.isEmpty()) {
            System.out.printf("Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE\n");
            return;
        }

        FpaTokenBuilder token = new FpaTokenBuilder();
        String result = token.buildToken(appId, appCertificate);
        System.out.printf("Token with FPA service: %s\n", result);
    }
}

File: DynamicKey/AgoraDynamicKey/kotlin/src/sample/kotlin/io/agora/sample/FpaTokenBuilderSample.kt
package io.agora.sample

import io.agora.media.FpaTokenBuilder

/** Demonstrates FPA Token007 generation. */
object FpaTokenBuilderSample {
    /** Generates and prints an FPA sample token. */
    @JvmStatic
    fun main(args: Array<String>) {
        val appId = System.getenv("AGORA_APP_ID")
        val appCertificate = System.getenv("AGORA_APP_CERTIFICATE")
        if (appId.isNullOrEmpty() || appCertificate.isNullOrEmpty()) {
            println("Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE")
            return
        }

        println("FPA token: ${FpaTokenBuilder().buildToken(appId, appCertificate)}")
    }
}
